### PR TITLE
Don't break OCC if an app is breaking in it's Application class

### PR DESCRIPTION
**Pull request by nickvergessen**
### Steps

1. Put the code from https://github.com/NastuzziSamy/files_external_gdrive/releases in place
2. Enable the app directly in the database, or comment out the `implements IBackendProvider` part in the Application class only for enabling
3. Try to disable it via OCC or do anything else in OCC

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

File: lib/private/AppFramework/Bootstrap/Coordinator.php [hunk 1]
```diff
@@ -113,21 +113,24 @@ private function registerApps(array $appIds): void {
...
-				try {
-					/** @var IBootstrap|App $application */
-					$apps[$appId] = $application = $this->serverContainer->query($applicationClassName);
-				} catch (QueryException $e) {
-					// Weird, but ok
-					continue;
-				}
-				try {
+			try {
+				if (class_exists($applicationClassName) && in_array(IBootstrap::class, class_implements($applicationClassName), true)) {
```
Comment (kesselb) [inline lib/private/AppFramework/Bootstrap/Coordinator.php:117]:
```
				if (!class_exists($applicationClassName) || !in_array(IBootstrap::class, class_implements($applicationClassName), true)) {
					continue;
				}
```

I find this version easier to read.

**Review by kesselb — APPROVED** [review]
LGTM :+1:

File: lib/private/AppFramework/Bootstrap/Coordinator.php (hunk 1 above)
Comment (nickvergessen) [reply lib/private/AppFramework/Bootstrap/Coordinator.php:117]: https://github.com/nextcloud/server/pull/26878/files?w=1 just retabing

File: lib/private/legacy/OC_App.php [hunk 2]
```diff
@@ -135,7 +135,14 @@ public static function loadApps(array $types = []): bool {
 		ob_start();
 		foreach ($apps as $app) {
 			if (!isset(self::$loadedApps[$app]) && ($types === [] || self::isType($app, $types))) {
-				self::loadApp($app);
+				try {
+					self::loadApp($app);
+				} catch (\Throwable $e) {
+					\OC::$server->get(LoggerInterface::class)->emergency('Error during app loading: ' . $e->getMessage(), [
```
Comment (MorrisJobke) [inline lib/private/legacy/OC_App.php:141]: This will then cause the app not being disabled anymore on errors, right? IMO a good thing 👍

**Pull request merged**